File: tests/end-to-end/apigatewayv2/typescript/stack.ts
import * as cdk from 'aws-cdk-lib';
import * as apigatewayv2 from 'aws-cdk-lib/aws-apigatewayv2';

export interface ApiGatewayV2StackProps extends cdk.StackProps {
}

export class ApiGatewayV2Stack extends cdk.Stack {
  /**
   * Endpoint for the HTTP API
   */
  public readonly apiEndpoint;

  public constructor(scope: cdk.App, id: string, props: ApiGatewayV2StackProps = {}) {
    super(scope, id, props);

    // Resources
    const myApi = new apigatewayv2.CfnApi(this, 'MyApi', {
      name: 'MyHttpApi',
      protocolType: 'HTTP',
      description: 'My HTTP API',
    });

    const defaultStage = new apigatewayv2.CfnStage(this, 'DefaultStage', {
      apiId: myApi.ref,
      stageName: 'default',
      autoDeploy: true,
    });

    const helloWorldIntegration = new apigatewayv2.CfnIntegration(this, 'HelloWorldIntegration', {
      apiId: myApi.ref,
      integrationType: 'HTTP_PROXY',
      integrationUri: 'https://jsonplaceholder.typicode.com/posts/1',
      integrationMethod: 'GET',
      payloadFormatVersion: '1.0',
    });

    const helloWorldRoute = new apigatewayv2.CfnRoute(this, 'HelloWorldRoute', {
      apiId: myApi.ref,
      routeKey: 'GET /hello',
      target: [
        'integrations',
        helloWorldIntegration.ref,
      ].join('/'),
    });

    // Outputs
    this.apiEndpoint = `https://${myApi.ref}.execute-api.${this.region}.amazonaws.com/default`;
    new cdk.CfnOutput(this, 'CfnOutputApiEndpoint', {
      key: 'ApiEndpoint',
      description: 'Endpoint for the HTTP API',
      value: this.apiEndpoint!.toString(),
    });
  }
}

File: tests/end-to-end/apigatewayv2/python/stack.py
from aws_cdk import Stack
import aws_cdk as cdk
import aws_cdk.aws_apigatewayv2 as apigatewayv2
from constructs import Construct

class ApiGatewayV2Stack(Stack):
  def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
    super().__init__(scope, construct_id, **kwargs)

    # Resources
    myApi = apigatewayv2.CfnApi(self, 'MyApi',
          name = 'MyHttpApi',
          protocol_type = 'HTTP',
          description = 'My HTTP API',
        )

    defaultStage = apigatewayv2.CfnStage(self, 'DefaultStage',
          api_id = myApi.ref,
          stage_name = 'default',
          auto_deploy = True,
        )

    helloWorldIntegration = apigatewayv2.CfnIntegration(self, 'HelloWorldIntegration',
          api_id = myApi.ref,
          integration_type = 'HTTP_PROXY',
          integration_uri = 'https://jsonplaceholder.typicode.com/posts/1',
          integration_method = 'GET',
          payload_format_version = '1.0',
        )

    helloWorldRoute = apigatewayv2.CfnRoute(self, 'HelloWorldRoute',
          api_id = myApi.ref,
          route_key = 'GET /hello',
          target = '/'.join([
            'integrations',
            helloWorldIntegration.ref,
          ]),
        )

    # Outputs
    """
      Endpoint for the HTTP API
    """
    self.api_endpoint = f"""https://{myApi.ref}.execute-api.{self.region}.amazonaws.com/default"""
    cdk.CfnOutput(self, 'CfnOutputApiEndpoint', 
      key = 'ApiEndpoint',
      description = 'Endpoint for the HTTP API',
      value = str(self.api_endpoint),
    )



